src/stream/src/executor/hash_join.rs

```diff
@@ -1252,6 +1438,93 @@ mod tests {
         (tx_l, tx_r, Box::new(executor).execute())
     }
 
+    #[tokio::test]
+    async fn test_interval_join() -> StreamExecutorResult<()> {
+        let chunk_l1 = StreamChunk::from_pretty(
+            "  I I
+             + 1 4
+             + 2 3
+             + 2 5
+             + 3 6",
+        );
+        let chunk_l2 = StreamChunk::from_pretty(
+            "  I I
+             + 3 8
+             - 3 8",
+        );
+        let chunk_r1 = StreamChunk::from_pretty(
+            "  I I
+             + 2 5
+             + 4 8
+             + 6 9",
+        );
+        let chunk_r2 = StreamChunk::from_pretty(
+            "  I  I
+             + 2 3
+             + 6 11",
+        );
+        let (mut tx_l, mut tx_r, mut hash_join) = create_executor::<{ JoinType::Inner }>(
+            true,
+            false,
+            Some(String::from("(equal:boolean $1:int8 $3:int8)")),
+            vec![(1, 3, true, None), (3, 1, true, None)],
+        )
```

Comment:
It seems there are should be inequality conditions here rather than an equal condition. Is it possible to fetch the row from the state store to ensure it has been deleted?

Reply:
> It seems there are should be inequality conditions here rather than an equal condition. Is it possible to fetch the row from the state store to ensure it has been deleted?

ok let me use band condition instead

Reply:
> > It seems there are should be inequality conditions here rather than an equal condition. Is it possible to fetch the row from the state store to ensure it has been deleted?
> 
> ok let me use band condition instead

done